psdk_wrapper/src/modules/telemetry.cpp

```diff
@@ -1253,6 +1268,70 @@ PSDKWrapper::altitude_barometric_callback(const uint8_t *data,
   return DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS;
 }
 
+T_DjiReturnCode
+PSDKWrapper::single_battery_index1_callback(const uint8_t *data,
+                                          uint16_t data_size,
+                                          const T_DjiDataTimestamp *timestamp)
+{
+  (void)data_size;
+  (void)timestamp;
+  std::unique_ptr<T_DjiFcSubscriptionSingleBatteryInfo> single_battery_info =
+      std::make_unique<T_DjiFcSubscriptionSingleBatteryInfo>(
+          *reinterpret_cast<const T_DjiFcSubscriptionSingleBatteryInfo *>(
+              data));
+
+  psdk_interfaces::msg::SingleBatteryInfo single_battery_info_msg;
+  single_battery_info_msg.header.stamp = this->get_clock()->now();
+
+  single_battery_info_msg.batteryindex = single_battery_info->batteryIndex;
+  single_battery_info_msg.voltage = static_cast<_Float32>(single_battery_info->currentVoltage) / 1000; // mV -> V
+  single_battery_info_msg.current = static_cast<_Float32>(single_battery_info->currentElectric) / 1000; // mA -> A
+  single_battery_info_msg.fullcapacity = static_cast<_Float32>(single_battery_info->fullCapacity) / 1000; // mAh -> Ah
+  single_battery_info_msg.capacityremain = static_cast<_Float32>(single_battery_info->remainedCapacity) /1000; // mAh -> Ah
+  single_battery_info_msg.temperature = static_cast<_Float32>(single_battery_info->batteryTemperature) / 10; // 0.1℃ -> ℃
+  single_battery_info_msg.cellcount = single_battery_info->cellCount;
+  single_battery_info_msg.selfcheckerror = single_battery_info->batteryState.selfCheckError;
+  single_battery_info_msg.closedreason = single_battery_info->batteryState.batteryClosedReason;
+  single_battery_info_msg.abnormalcomm = single_battery_info->batteryState.batteryCommunicationAbnormal;
+  single_battery_info_msg.isembed = single_battery_info->batteryState.isBatteryEmbed;
+  single_battery_index1_pub_->publish(single_battery_info_msg);
+  return DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS;
+
+}
+
+T_DjiReturnCode
+PSDKWrapper::single_battery_index2_callback(const uint8_t *data,
+                                          uint16_t data_size,
+                                          const T_DjiDataTimestamp *timestamp)
+{
+  (void)data_size;
+  (void)timestamp;
+  std::unique_ptr<T_DjiFcSubscriptionSingleBatteryInfo> single_battery_info =
+      std::make_unique<T_DjiFcSubscriptionSingleBatteryInfo>(
+          *reinterpret_cast<const T_DjiFcSubscriptionSingleBatteryInfo *>(
+              data));
+
+  psdk_interfaces::msg::SingleBatteryInfo single_battery_info_msg;
+  single_battery_info_msg.header.stamp = this->get_clock()->now();
+
+  single_battery_info_msg.batteryindex = single_battery_info->batteryIndex;
```

Comment:
why are we not using camel case here? so, wouldn't it be more clear to have battery_index, full_capacity and not all together? Is there a reason for this?

Reply:
no there is not a reason for that, I already pushed a commit with these changes, you can take a look